integration_tests/traffic_test.go

```diff
@@ -0,0 +1,291 @@
+	rxPkts := otg.Telemetry().Flow("Flow").Counters().InPkts().Get(t)
+	lossPct := (txPkts - rxPkts) * 100 / txPkts
+	return float32(lossPct)
+}
+
+// awaitTimeout calls a fluent client Await, adding a timeout to the context.
+func awaitTimeout(ctx context.Context, c *fluent.GRIBIClient, t testing.TB, timeout time.Duration) error {
+	subctx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+	return c.Await(subctx, t)
+}
+
+// TestIPv4Entry tests a single IPv4Entry forwarding entry.
+func TestIPv4Entry(t *testing.T) {
+	ctx := context.Background()
+
+	dut := ondatra.DUT(t, "dut")
+	configureDUT(t, dut)
+
+	ate := ondatra.ATE(t, "ate")
+	ateTop := configureATE(t, ate)
+	ate.OTG().PushConfig(t, ateTop)
+
+	cases := []struct {
+		desc                 string
+		entries              []fluent.GRIBIEntry
+		wantOperationResults []*client.OpResult
+	}{{
+		desc: "Single next-hop",
+		entries: []fluent.GRIBIEntry{
+			fluent.NextHopEntry().WithNetworkInstance("DEFAULT").
+				WithIndex(nhIndex).WithIPAddress(atePort2.IPv4),
+			fluent.NextHopGroupEntry().WithNetworkInstance("DEFAULT").
+				WithID(nhgIndex).AddNextHop(nhIndex, 1),
+			fluent.IPv4Entry().WithNetworkInstance("DEFAULT").
+				WithPrefix(ateDstNetCIDR).WithNextHopGroup(nhgIndex),
+		},
+		wantOperationResults: []*client.OpResult{
+			fluent.OperationResult().
+				WithNextHopOperation(nhIndex).
+				WithProgrammingResult(fluent.InstalledInFIB).
+				WithOperationType(constants.Add).
+				AsResult(),
+			fluent.OperationResult().
+				WithNextHopGroupOperation(nhgIndex).
+				WithProgrammingResult(fluent.InstalledInFIB).
+				WithOperationType(constants.Add).
+				AsResult(),
+			fluent.OperationResult().
+				WithIPv4Operation(ateDstNetCIDR).
+				WithProgrammingResult(fluent.InstalledInFIB).
+				WithOperationType(constants.Add).
+				AsResult(),
+		},
+	}}
+	for _, tc := range cases {
+		t.Run(tc.desc, func(t *testing.T) {
+			gribic := dut.RawAPIs().GRIBI().Default(t)
+			c := fluent.NewClient()
+			c.Connection().WithStub(gribic).
+				WithRedundancyMode(fluent.ElectedPrimaryClient).
+				WithPersistence().
+				WithFIBACK().
+				WithInitialElectionID(1, 0)
+			c.Start(ctx, t)
+			defer c.Stop(t)
+			c.StartSending(ctx, t)
+			if err := awaitTimeout(ctx, c, t, time.Minute); err != nil {
+				t.Fatalf("Await got error during session negotiation: %v", err)
+			}
+
+			c.Modify().AddEntry(t, tc.entries...)
+			if err := awaitTimeout(ctx, c, t, time.Minute); err != nil {
+				t.Fatalf("Await got error for entries: %v", err)
+			}
+			defer func() {
+				c.Modify().DeleteEntry(t, tc.entries...)
+				if err := awaitTimeout(ctx, c, t, time.Minute); err != nil {
+					t.Fatalf("Await got error for entries: %v", err)
+				}
+			}()
+
+			for _, wantResult := range tc.wantOperationResults {
+				chk.HasResult(t, c.Results(t), wantResult, chk.IgnoreOperationID())
+			}
+
+			loss := testTraffic(t, ate, ateTop, atePort1, atePort2)
+			if loss > 10 {
```

Comment:
What's the traffic loss you're seeing? I think we want something less than 10.

Reply:
added TODO for this, if the neighbor table is warmed up packet loss is < 1% if it isn't loss is high

Reply:
actually looking at the logs it's the configured interfaces that's slow, trying with Await()

Reply:
this looks much better after waiting for IPs to be configured.